tooling/ssa_fuzzer/fuzzer/src/fuzz_lib/function_context.rs

```diff
@@ -22,15 +22,66 @@ const NUMBER_OF_BLOCKS_INSERTING_IN_JMP: usize = 1;
 const NUMBER_OF_BLOCKS_INSERTING_IN_JMP_IF: usize = 2;
 const NUMBER_OF_BLOCKS_INSERTING_IN_LOOP: usize = 4;
 
+pub(crate) type ValuesTypes = (Vec<FieldElement>, Vec<ValueType>);
+
+/// Field modulus has 254 bits, and FieldElement::from supports u128, so we use two unsigneds to represent a field element
+/// field = low + high * 2^128
+#[derive(Debug, Clone, Copy, Hash, Arbitrary, Serialize, Deserialize)]
+pub(crate) struct FieldRepresentation {
+    pub(crate) high: u128,
+    pub(crate) low: u128,
+}
+
+impl From<&FieldRepresentation> for FieldElement {
+    fn from(field: &FieldRepresentation) -> FieldElement {
+        let lower = FieldElement::from(field.low);
+        let upper = FieldElement::from(field.high);
+        lower + upper * (FieldElement::from(u128::MAX) + FieldElement::from(1_u128))
+    }
+}
+
+#[derive(Debug, Clone, Copy, Hash, Arbitrary, Serialize, Deserialize)]
+pub(crate) enum WitnessValue {
+    Field(FieldRepresentation),
+    U64(u64),
+    Boolean(bool),
+    I64(u64),
+    I32(u32),
+}
+
+/// TODO(sn): initial_witness should be in ProgramData
+/// Represents the data describing a function
+#[derive(Arbitrary, Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct FunctionData {
+    pub(crate) blocks: Vec<InstructionBlock>,
```

Comment:
I think blocks should be global and not function-specific. And then you construct functions out of them through commands